src/backend/joanie/core/models/products.py

```diff
@@ -377,24 +386,32 @@ class Meta:
     def __str__(self):
         return f"Order {self.product} for user {self.owner}"
 
-    @property
-    def state(self):
+    def can_be_state_validated(self):
```

Comment:
I think we are doing the reverse as what we should do here.
The order is created
The order exposes a `validate` action on its detail endpoint.
Once the frontend is ready to confirm the order, it calls the `validate` action.
then it initiates the payment on this validated order.
once the transaction is received from the webhook, we see that the invoice is paid so wet transition the order to `paid` state.

Reply:
that wouldn't be ISO though with what exists in the code today

Reply:
As discussed IRL, let's open an issue and merge it like it is so that we don't postpone going to production!